docs/src/guides/architecture-patterns.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Architecture Patterns
+navTitle: Architecture
+description: Design Patterns in Hub.js
+order: 1
+group: 2-concepts
+---
+
+# Architecture Patterns in Hub.js
+
+Hub.js has evolved overs a number of years, and has a number of developer "audiences". As such, there are three broad Architectural Patterns employed: Plain Functions, Manager Classes, and the Hub Automation Class.
+
+## Plain Functions
+
+As it's core, Hub.js provides a collection of functions for working with ArcGIS Hub entities (Sites/Pages/Initiatives/Projects etc), and/or streamlining common Hub related workflows. These are designed to be flexibly composed/orchestrated by applications.
+
+While this pattern enables the wide reuse, and simple composition, it comes at the cost of increased cognative load for developers - you need to know what functions to use and what context to use them in. The [API Reference](/hub.js/api/) contains a full list of all the functions.
+
+Additional abstractions discussed below are composed from this foundational pattern, always allowing a developer to opt into a lower-level of abstraction, enabling composition for unforseen workflows. Additionally, working with modern build tools, this level of abstraction will result in the smallest built output, as functions are the simplest structures to [tree shake](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/) during a build (assuming you only import the functions your code needs!).
+
+## Manager Classes
+
+To reduce cognative load, and make it easier for developers outside the ArcGIS Hub team, we have introduced a set of Manager Classes. These classes are Entity Specific (i.e. there is a `HubProjectManager` class), and they encapsulate the standard operations for an Entity - `create`, `update`, `destroy`, `fetch` etc, as well as entity specific operations.
+
+These classes are instantiated as needed, using either an `ArcGISContext` or an `ArcGISContextManager` to provide platform and identity information. The [`ArcGISContextManager` documentation](./context.html) has additional background information.
```

Comment:
This is confusing as the focus shifts to another concept (context) without example of the primary topic of Managers - show how a Manager is created.

Reply:
Agreed that this is complex, but in order to instantiate a manager, a contextManager or `IHubContext` need to be passed in. Thus I described this first, then showed in the code